messages.go

```diff
@@ -391,6 +392,53 @@ func (ppcm *ProofPCommitmentMap) MarshalJSON() ([]byte, error) {
 	return json.Marshal(encPPCM)
 }
 
+type PMap struct {
+	Ps map[PublicKeyIdentifier]*big.Int `json:"p"`
+}
+
+func (pm *PMap) MarshalJSON() ([]byte, error) {
+	var encPM struct {
+		Ps map[string]*big.Int `json:"p"`
+	}
+	encPM.Ps = make(map[string]*big.Int)
+
+	for pki, v := range pm.Ps {
+		pkiBytes, err := pki.MarshalText()
+		if err != nil {
+			return nil, err
+		}
+		encPM.Ps[string(pkiBytes)] = v
+	}
+
+	return json.Marshal(encPM)
+}
+
+type GetCommitmentsRequest struct {
+	Keys []PublicKeyIdentifier          `json:"keys"`
+	Hash gabi.KeyshareCommitmentRequest `json:"hw"`
+}
+
+type ProofPCommitmentMapV2 struct {
+	Commitments map[PublicKeyIdentifier]*big.Int `json:"c"`
+}
+
+func (cm *ProofPCommitmentMapV2) MarshalJSON() ([]byte, error) {
+	var encCM struct {
+		Commitments map[string]*big.Int `json:"p"`
+	}
```

Comment:
Shouldn't the `Commitments` field have the JSON tag `c`?

Reply:
yep, seems a copy-paste mistake. i'll fix it